_LowLevelCode/cpp/CommonCode.h

```diff
@@ -75,7 +72,24 @@ inline void copy_pixels(double* pixel_data, long j, double* pPixelSorted, size_t
 
 
 }
-
+//* Possible prototype for a generic function
+template<class T>
```

Comment:
This function exists: https://uk.mathworks.com/help/matlab/apiref/mxgetscalar.html

Reply:
thous would return real*8 scalar, where template is applicable for `real*4` and `uint64_t` -- I suspect I may consider these for some reasons.